### PR TITLE
only import ssl if it's used

### LINKED ISSUE #1019: Make ssl an optional import
Not everyone wants to (or can, for license reasons) use openssl.  It looks like we import ssl at https://github.com/jupyter/notebook/blob/a3e074c3a7e29eb3ec8b94627370698458f048ce/notebook/notebookapp.py#L22, but the *only* place the import is used is at https://github.com/jupyter/notebook/blob/a3e074c3a7e29eb3ec8b94627370698458f048ce/notebook/notebookapp.py#L873 and https://github.com/jupyter/notebook/blob/a3e074c3a7e29eb3ec8b94627370698458f048ce/notebook/notebookapp.py#L879, both just for the constants.

Thanks to @williamstein for making us aware of the issue.

### PR DESCRIPTION
**Pull request by minrk**
SSL stdlib module may be unavailable in rare cases.

closes #1019

### CODE REVIEW

File: notebook/notebookapp.py [hunk 1]
```diff
@@ -870,13 +868,17 @@ def init_webapp(self):
             ssl_options['keyfile'] = self.keyfile
         if self.client_ca:
```
Comment (jasongrout) [inline notebook/notebookapp.py]: Is there a reason this entire if statement isn't moved down below in the else block?

Comment (jasongrout) [inline notebook/notebookapp.py]: never mind, figured the logic out.

**Comment by jasongrout** [comment]
Travis errors are for an unrelated bug that appears to be fixed in master.  Thanks, Min!

**Comment by Carreau** [comment]
Marking as 4.2 to think about backporting.

**Pull request merged**